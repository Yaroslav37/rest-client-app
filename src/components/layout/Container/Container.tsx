interface Props {
  children: React.ReactNode;
  className?: string;
}

export const Container = ({ children, className }: Props) => {
  return (
    <div className={`px-4 sm:px-6 xl:px-10 max-w-[1920px] mx-auto ${className}`}>{children}</div>
  );
};
