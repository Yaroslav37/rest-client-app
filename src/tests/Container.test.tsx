import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { Container } from '@/components/layout/Container/Container';

describe('Container Component', () => {
  it('renders children correctly', () => {
    const { getByText } = render(
      <Container>
        <div>Test Content</div>
      </Container>,
    );

    expect(getByText('Test Content')).toBeInTheDocument();
  });

  it('applies default classes correctly', () => {
    const { container } = render(
      <Container>
        <div>Test</div>
      </Container>,
    );

    const div = container.firstChild;
    expect(div).toHaveClass('px-4');
    expect(div).toHaveClass('sm:px-6');
    expect(div).toHaveClass('xl:px-10');
    expect(div).toHaveClass('max-w-[1920px]');
    expect(div).toHaveClass('mx-auto');
  });

  it('merges custom className with default classes', () => {
    const { container } = render(
      <Container className="custom-class extra-class">
        <div>Test</div>
      </Container>,
    );

    const div = container.firstChild;
    expect(div).toHaveClass('px-4');
    expect(div).toHaveClass('sm:px-6');
    expect(div).toHaveClass('custom-class');
    expect(div).toHaveClass('extra-class');
  });

  it('matches snapshot with default props', () => {
    const { container } = render(
      <Container>
        <div>Test</div>
      </Container>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with custom className', () => {
    const { container } = render(
      <Container className="custom-class">
        <div>Test</div>
      </Container>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
