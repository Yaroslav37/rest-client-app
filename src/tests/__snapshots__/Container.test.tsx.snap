// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Container Component > matches snapshot with custom className 1`] = `
<div
  class="px-4 sm:px-6 xl:px-10 max-w-[1920px] mx-auto custom-class"
>
  <div>
    Test
  </div>
</div>
`;

exports[`Container Component > matches snapshot with default props 1`] = `
<div
  class="px-4 sm:px-6 xl:px-10 max-w-[1920px] mx-auto undefined"
>
  <div>
    Test
  </div>
</div>
`;
